.github/workflows/packaging-test.yml

```diff
@@ -2,6 +2,9 @@ name: Packaging Tests
 
 on:
   pull_request:
+  # Run daily at 1:23 UTC
+  schedule:
+    - cron: '23 1 * * *'
```

Comment:
So something like

```yaml
on:
  push:
    branches:
    - main
  # Run daily at 1:23 UTC
  schedule:
    - cron: '23 1 * * *'
  workflow_dispatch:
```

?

Reply:
I am on mobile and coming down with a cold, so not super speedy of thought right now! What I'm proposing is to only run full blown packaging tests on main, if we want to save the cycles associated with it.

If Jim doesn't feel the need to do this, then we should definitely add the cron as you have!

Reply:
Ah gotcha. Yeah, this is what I did for `matplotlib` (https://github.com/matplotlib/matplotlib/blob/c2aa4ee1177452dc778813e3034c3f57d7db83d8/.github/workflows/nightlies.yml) so I'm happy to change it to be `main` and `push` based.

Reply:
We absolutely need to keep testing commits on PR branches, and it is the case that `main` gets tested after a merge. (I don't know which part of the CI does this, but I've observed that it happens every time, with the way we have things set up.)

I guess it's advantageous to test on a regular interval, to catch when third-party updates break things, and daily is the right cadence for that. So sure, let's add daily testing on `main`.

This is in addition to (or part of) the process that makes daily artifacts for Anaconda Cloud?

Reply:
@jpivarski I don't think we explicitly test main, apart from docs right now. Our PR tests are evaluated against the merge commit with main due to the checkout action (I am pretty sure), so it's fairly safe.

Reply:
>  it is the case that main gets tested after a merge. (I don't know which part of the CI does this, but I've observed that it happens every time, with the way we have things set up.)
> 
> I guess it's advantageous to test on a regular interval, to catch when third-party updates break things, and daily is the right cadence for that. So sure, let's add daily testing on main.

If there is already instances in which there are runs that produce wheels on `main` then that's really enough, and the nightly wheels are more about capturing the state of dev at `HEAD`, so let me revert what I just added and then look at what workflows run on pushes to `main` from merges.

Reply:
Actually no wheels are built an published on main. We do build the single platform for docs, but that's not using cibuildwheel.

Reply:
Yeah, looking at `push` events on `main` (https://github.com/scikit-hep/awkward/actions?query=event%3Apush+branch%3Amain) this shows what @agoose77 just mentioned

![image](https://github.com/scikit-hep/awkward/assets/5142394/4b4ab755-43f0-4e55-adf9-f1227466c714)

Reply:
I was referring to testing, not wheel-publishing. I've seen tests on `main` after each PR merge, but I take it that the wheel-building/publishing would be a new step.

Reply:
Ah, you were only ever talking about wheel building/publishing. Yes, we can add that on `main` as a new feature.

Reply:
Just to make the point for clarity (I am sure we are all on the same page), we don't really need to test main given that we're testing the merge commit. I'm pretty certain that's bullet proof.

As such, we don't need the packaging tests in PRs, as we build wheels for all (?) platforms. However, the packaging test ensures our cibuildwheel works, which isn't otherwise tested until deployment. 

But we don't want to build nighties on every pr, just the ones that made it to main. So to me, that means building main.

Reply:
> But we don't want to build nighties on every pr, just the ones that made it to main. So to me, that means building main.

Yes, this is what we did for `matplotlib`. But at the moment the _full_ wheel building, as opposed to the subset of wheel build tests, is only run on deployment.

https://github.com/scikit-hep/awkward/blob/dd4753b97949e4098aa6008a189624ba7a72fda6/.github/workflows/deploy.yml#L1-L11

So what would be better would be to split the wheel building out into a seperate workflow and then have that workflow get called during publishing. Idk if you want to do that in a seperate PR though.

> Just to make the point for clarity (I am sure we are all on the same page), we don't really need to test main given that we're testing the merge commit. I'm pretty certain that's bullet proof.

Right. So I'm going to revert the last commit to where things were before we got talkinga bout this.